projects/plugins/boost/app/modules/cache/file_cache.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+require_once __DIR__ . '/cache.php';
+
+class Boost_File_Cache extends Boost_Cache {
+	private $path           = false;
+	private $cache_filename = false;
+
+	private function calculate_path( $request_uri ) {
+		$key  = md5( $request_uri );
+		$path = WP_CONTENT_DIR . DIRECTORY_SEPARATOR . 'boost-cache' . DIRECTORY_SEPARATOR . 'cache' . DIRECTORY_SEPARATOR;
+		for ( $i = 0; $i <= 5; $i++ ) {
+			$path .= substr( $key, $i, 1 ) . DIRECTORY_SEPARATOR;
+		}
+
+		return $path;
+	}
+
+	private function path( $request_uri = false ) {
+		if ( $request_uri !== false ) {
+			return $this->calculate_path( $request_uri );
+		}
+		if ( $this->path === false ) {
+			$this->path = $this->calculate_path( $this->request_uri );
+		}
+		return $this->path;
+	}
+
+	private function cache_filename() {
+		if ( $this->cache_filename === false ) {
+			$this->cache_filename = $this->path() . $this->cache_key() . '.php';
```

Comment:
If we're saving all cache files as `.php`, could it ever be possible for a user to inject a malicious comment or something which displays `<?php` tags in the page output, and then execute it by directly calling the relevant `.php` file from the cache directory?

Do we need to prepend a `<?php\ndie();\n` to every cached file for safety? Would it make sense to cache them with `.html` on the end?

Reply:
You're right. I was thinking of adding meta data like headers, and canonical URL to this file, but it could be added as a meta sidecar file, just as easily. In WP Super Cache, all cache files are .html while meta files are .php and do have a `die()` at the start of each file.